include/deal.II/lac/cuda_kernel.h

```diff
@@ -0,0 +1,554 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_cuda_kernel_h
+#define dealii_cuda_kernel_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/cuda_size.h>
+
+#include <deal.II/lac/cuda_atomic.h>
+#include <deal.II/lac/cuda_vector.h>
+
```

Comment:
Does this not need an `#ifdef DEAL_II_WITH_CUDA` here, as is done in `lac/cuda_vector.cu`?

Reply:
Yes, I actually cherry-pick this from another branch I am working on and I have added the preprocessor guard in another commit. Thanks.